### PR TITLE
Update template for 0.6.0

### PR DESCRIPTION
**Pull request by jeddenlea**
The API is changing so our default template needs some adjustments.
There are also a small number of inconsistencies to the template that we
can address at the same time.

First and foremost, this simply adjusts the intent of the template to
use the new API.

Second, to enable URL purging, apps must accept and forward "PURGE"
requests, so this now does so.

But last, the routing logic needs a little bit of cleanup.  While one
section of the code suggested that HEAD and POST requests were accepted,
they would later all be issued 404 responses.  The vast majority of
Fastly services have no real need to support POST requests, so we
shouldn't imply they should in our blessed example templates.

Differentiating GETs and HEADs can potentially get tricky, but in this
simple example, we may simply treat them as though they are the same.
If a Response with a body is made to a HEAD request, the body will be
dropped, but its Content-Length will be sent.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -8,4 +8,5 @@ edition = "2018"
 debug = true
 
 [dependencies]
-fastly = "^0.5.0"
+# XXX: set to release version
+fastly = "^0.6.0-beta3"
```
Comment (cratelyn) [inline Cargo.toml:12]: This can be `^0.6.0` now, but I believe I'm stating the obvious here.

**Review by cratelyn — APPROVED** [review]
Thank you!!

**Comment by cratelyn** [comment]
I believe CI is failing because it's still using 1.46 somewhere?

**Comment by JakeChampion** [comment]
> I believe CI is failing because it's still using 1.46 somewhere?

Yep, in the test github workflow file on line 7:
https://github.com/fastly/compute-starter-kit-rust-default/blob/4f2c8a2a4e4dd3c81676dd8733d30bb83b5126ac/.github/workflows/test.yml#L7

**Comment by cratelyn** [comment]
I'm going to close this, as this work is continuing onward in #28 :slightly_smiling_face:

**Pull request closed without merging**